functions/anonymous-functions.md

```diff
@@ -0,0 +1,198 @@
+  console.log('Inside this anonymous function, each item of myArray will be accessed as item:', item);
+});
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                                    |
+| --------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
+| `const myArray = [...];`                            | Here we create an array, `myArray`.                                                                                                      |
+| `myArray.forEach(...);`                             | To iterate over each element in `myArray`, we call the `forEach` function off `myArray`.                                                 |
+| `item => { ... }`                                   | The `forEach` function takes in one argument: another function! This anonymous function is using arrow function syntax.                  |
+| `item`                                              | The anonymous function we pass into `forEach` should take in one argument itself. This argument will always represent the iterated item. |
+| `console.log(..., item);`                           | In the function body, we can access the value of `item` and read it!                                                                     |
+
+Anonymous functions can be arrow functions or non-arrow functions! Whether we use an arrow function or non-arrow function depends on the context of the problem, and which form of `this` we need.
+
+### Example: Iterating Through Tips
+
+Let's look at another example! Here, we have an array of cash tips, `cashTips`. We want to find the average cash tip. We will create a `sum` variable. Then, we'll iterate through `cashTips`. For each tip in the array, we'll increment `sum`. Then we'll calculate the average cash tip.
+
+```javascript
+const cashTips = [4, 7, 9, 12, 3, 18, 6];
+let sum = 0;
+
+cashTips.forEach((tip) => {
+  sum += tip;
+});
+
+const average = sum / cashTips.length;
+console.log(`The average cash tip is ${average}`);
+```
+
+To create the loop, we call `cashTips.forEach()`. The `forEach` function takes in one argument: another function.
+
+Here, we pass in an anonymous function:
+
+<!-- prettier-ignore-start -->
+```js
+tip => {
+  sum += tip;
+}
+```
+<!-- prettier-ignore-end -->
+
+This function takes in one argument, `tip`. Our `forEach` function will use this anonymous function, and each `tip` will be each item in `cashTips`. Inside our anonymous function, we increment `sum` by `tip`.
+
+### Anonymous Functions in Context
+
+This is a great example of an anonymous function because:
+
+- The logic of this function isn't reused anywhere
+- The function is more readable in the context of the `forEach` loop than outside it
+
```

Comment:
include a callout hinting at closures and how anonymous functions (our definition) can use values from the enclosing scope
```suggestion

### !callout-info

## The Curious Case of `sum`

Was it surprising that we could access `sum` from within our anonymous function? Here, we can explain our ability to see `sum` because `sum` has global scope. However, due to a language feature called _closure binding_, even if `sum` had been a local variable in some local scope, it would _still_ have been accessible in our anonymous function!

<br />

This is another benefit of using an anonymous function defined here rather than a named function defined elsewhere. A function defined elsewhere wouldn't have access to the variables in the local scope around the `forEach` call. But our anonymous function does!

### !end-callout

```

Reply:
I think I'd like the title of this to feel a little more "aside-y" and also name "Closure Binding"

```suggestion

### !callout-info

## JavaScript Details: `sum` and Closure Binding

Was it surprising that we could access `sum` from within our anonymous function? Here, we can explain our ability to see `sum` because `sum` has global scope. However, due to a language feature called _closure binding_, even if `sum` had been a local variable in some local scope, it would _still_ have been accessible in our anonymous function!

<br />

This is another benefit of using an anonymous function defined here rather than a named function defined elsewhere. A function defined elsewhere wouldn't have access to the variables in the local scope around the `forEach` call. But our anonymous function does!

### !end-callout

```